prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1782,6 +1790,7 @@ layoutWadlerLeijen
         -> SimpleDocStream ann -- ^ Choice B.
         -> SimpleDocStream ann -- ^ Choice A if it fits, otherwise B.
     selectNicer (FittingPredicate fits) lineIndent currentColumn x y
+      | isFail x = y
```

Comment:
I was wondering whether we need to worry about `SFail` not appearing at the beginning of a `Union`. At least `fuse` doesn't seem to "hide" the `Fail` in a  `(Union Fail x)` though:

https://github.com/quchen/prettyprinter/blob/4d51362d7ea8bca0fe9e7ffec2ccd8fcd0065c0d/prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs#L1407

Reply:
`isFail` is `False` if the document fails, but that’s the same thing the two fitting predicates (for `layoutPretty` and `layoutSmart`) yield. What does the explicit `isFail` clause change?

Reply:
I've explained this in https://github.com/quchen/prettyprinter/issues/91#issuecomment-549033715.

The short version is that both fitting predicates shortcut to `True` when the `maxWidth` is `Nothing` – which is the case with `Unbounded` layout options.